ui/src/components/Database/Aside/components/DialogResources/TagDialog.vue

```diff
@@ -0,0 +1,168 @@
+<!--
+  ~ Licensed to Apache Software Foundation (ASF) under one or more contributor
+  ~ license agreements. See the NOTICE file distributed with
+  ~ this work for additional information regarding copyright
+  ~ ownership. Apache Software Foundation (ASF) licenses this file to you under
+  ~ the Apache License, Version 2.0 (the "License"); you may
+  ~ not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+-->
+
+<script setup>
+import { watch, ref, reactive } from 'vue'
+const rule = {
+    tagFamily: [
+        {
+            required: true, message: 'Please enter the tagFamily name', trigger: 'blur'
+        }
+    ]
+}
+const props = defineProps({
+    visible: {
+        type: Boolean,
+        default: false,
+        required: true
+    },
+    operation: {
+        type: String,
+        default: 'Add',
+        required: true
+    },
+    type: {
+        type: String,
+        default: 'Tag',
+        required: true
+    },
+    group: {
+        type: String,
+        required: true
+    },
+    name: {
+        type: String,
+        required: true
+    }
+})
+
+let data = reactive({
+    dialogVisible: false,
+    form: {
+        metadata: {
+            group: '',
+            name: '',
+            tagFamily: ''
+        }
+    }
+})
+let rules = rule
+let tags = [{
+    name: 'start_time',
+    type: 'TAG_TYPE_INT',
+    indexedOnly: false
+}]
+const ruleForm = ref()
+const emit = defineEmits(['confirm', 'cancel'])
+
+watch(() => props.visible, () => {
+    data.dialogVisible = props.visible
+})
+watch(() => props.group, () => {
+    data.form.metadata.group = props.group
+})
+
+function confirmForm() {
+    ruleForm.value.validate((valid) => {
+        if (valid) {
+            emit('confirm', data.form)
```

Comment:
Could you add validation to check the tag's name is unique in all tag families? And the tag family's name can not be duplicated as well.

Reply:
of course